packages/@react-spectrum/s2/src/SegmentedControl.tsx

```diff
@@ -97,7 +106,13 @@ const controlItem = style({
 }, getAllowedOverrides())
 
 interface SegmentedControlInternalContextProps extends SegmentedControlProps {
-  register?: (value) => void;
+  register?: (value: string) => void;
```

Comment:
I think we generally follow the pattern of prefix the internal context with Internal https://github.com/adobe/react-spectrum/blob/main/packages/%40react-spectrum/s2/src/Picker.tsx#L219
It makes it a little easier to skim for